### ISSUE TITLE
Wrong display bounds exposed in Wayland but not in Xwayland

### ISSUE THREAD

**Issue by EmmanuelARB** [swaywm/sway#6830]
Hi,

Platform : Arch Linux on kernel x86_64 5.16.5-arch1-1
CPU/GPU : Intel(R) Core(TM) i5-7200U CPU
Sway versions : both 1.7-2 and latest git (with both wlroots 0.15.1-2 and latest git)

I have a weird issue I encountered on SDL apps where the wrong bounds are exposed when using Wayland but not on Xwayland.
I initially posted the issue on the SDL github ( https://github.com/libsdl-org/SDL/issues/5311  ) but it seems to be compositor-specific and I'm not sure where to start looking to find the real culprit.

Basically, the exact same program outputs on Xwayland 
> INFO: 0, 2560, 210
> INFO: 1, -1080, -70
> INFO: 2, 0, 0
(correct answer)

and on Wayland
> INFO: 0, 0, 0
> INFO: 1, 0, 0
> INFO: 2, 0, 0
(obviously wrong)

This is causing issues in some program, for instance Moonlight-QT, which always starts on the wrong display.

The issue is actually quite new, a couple of months old at most, and before it was working quite well.

Thanks a lot for any help pointing me towards the right direction !
Emmanuel

**Comment by emersion** [comment]
On Wayland clients don't decide their global position. Sway always sends 0,0 for all output positions because clients shouldn't care.

**Comment by EmmanuelARB** [comment]
Thanks for that quick answer ! I quite understand the logic but I still find it misleading, especially since other compositors are not taking the same approach which can lead to different behaviors.

I'll try to work it around on the only app I found where this lead to unintended behavior (Moonlight-QT).

**Comment by cgutman** [comment]
@emersion The use case here is trying to put the game window on the same screen that the user was using for the configuration window, since users tend not to like windows flying around to different screens.

Do you have an alternate strategy to achieve this?
